src/fmpz/link/fmpz_reentrant.c

```diff
@@ -13,17 +14,29 @@
 #include "gmpcompat.h"
 #include "fmpz.h"
 
-__mpz_struct * _fmpz_new_mpz(void)
+__GMP_DECLSPEC extern void * (*__gmp_allocate_func) (size_t);
+__GMP_DECLSPEC extern void * (*__gmp_reallocate_func) (void *, size_t, size_t);
+__GMP_DECLSPEC extern void   (*__gmp_free_func) (void *, size_t);
+#define ALLOC(x) (x)->_mp_alloc
+#define SIZ(x) (x)->_mp_size
+#define PTR(x) (x)->_mp_d
+
+mpz_ptr _fmpz_new_mpz2(mp_size_t limbs)
 {
-    __mpz_struct * mf = (__mpz_struct *) flint_malloc(sizeof(__mpz_struct));
-    mpz_init2(mf, 2*FLINT_BITS);
+    mpz_ptr mf = (mpz_ptr) flint_malloc(sizeof(__mpz_struct));
+    /* mpz_init2(mf, 2*FLINT_BITS); */
+    ALLOC(mf) = 2;
```

Comment:
The 2 doesn't match the number of limbs that is actually allocated.

Reply:
Thanks